cmd/kool_task.go

```diff
@@ -30,40 +35,119 @@ func (t *DefaultKoolTask) Run(args []string) (err error) {
 		return t.Execute(args)
 	}
 
-	var output string
+	t.taskOut.SetWriter(t.GetWriter())
+	pipeReader, pipeWriter := io.Pipe()
 
-	t.Printf("%s ... ", t.message)
+	t.SetWriter(pipeWriter)
+	defer t.SetWriter(t.taskOut.GetWriter())
 
-	originalWriter := t.GetWriter()
-	buf := bytes.NewBufferString("")
-	t.SetWriter(buf)
+	startMessage := fmt.Sprintf("%s ...", t.message)
+	t.taskOut.Println(startMessage)
+	t.taskOut.Println(strings.Repeat("=", len(startMessage)))
+
+	lines := make(chan string)
+
+	doneScanning := startServiceOutputScanner(pipeReader, lines)
+	donePrinting := t.startServiceOutputPrinter(lines, doneScanning)
 
 	err = <-t.execService(args)
+	pipeWriter.Close()
 
-	t.SetWriter(originalWriter)
-	bufBytes, _ := ioutil.ReadAll(buf)
-	output = strings.TrimSpace(string(bufBytes))
+	<-doneScanning
+	<-donePrinting
 
+	var statusMessage string
 	if err != nil {
-		t.Println(color.New(color.Red).Sprint("error"))
+		statusMessage = fmt.Sprintf("... %s", color.New(color.Red).Sprint("error"))
 	} else {
-		t.Println(color.New(color.Green).Sprint("done"))
+		statusMessage = fmt.Sprintf("... %s", color.New(color.Green).Sprint("done"))
 	}
 
-	if output != "" {
-		t.Println(output)
-	}
+	t.taskOut.Printf("\r")
+	t.taskOut.Println(statusMessage)
 
 	return
 }
 
 func (t *DefaultKoolTask) execService(args []string) <-chan error {
-	chError := make(chan error)
+	err := make(chan error)
+
+	go func() {
+		defer close(err)
+		err <- t.Execute(args)
+	}()
+
+	return err
+}
+
+func startServiceOutputScanner(reader io.Reader, lines chan string) <-chan bool {
+	doneScanning := make(chan bool)
+	scanner := bufio.NewScanner(reader)
+	scanner.Split(bufio.ScanLines)
+
+	go func() {
+		defer func() {
+			close(lines)
+			close(doneScanning)
+		}()
+
+		for range time.Tick(100 * time.Millisecond) {
+			if !scanner.Scan() {
```

Comment:
let's check how it goes if we do `for scanner.Scan() ` instead of using the time.Tick above for throttling.

Reply:
we need to be aware of possible CPU high usage

Reply:
https://golang.org/pkg/bufio/#Reader.ReadString could be an option too